lib/task.js

```diff
@@ -97,15 +98,20 @@ function factory(
         self.successStates = [TaskStates.Succeeded];
         // hint to whatever is running the task when it has failed
         self.failedStates = [TaskStates.Failed, TaskStates.Timeout, TaskStates.Cancelled];
-
+   
+        var server;
+        if (_.has(context, 'proxy')) {
+            server = context.proxy;
+        } else {
+            server = 'http://%s:%s'.format(
+                Task.configCache.apiServerAddress,
+                Task.configCache.apiServerPort
```

Comment:
I'm afraid these two configuration entries may be deprecated, because we have switch them to the southbound API endpoint, see the example configuration (http://rackhd.readthedocs.io/en/latest/packaging_and_deployment.html#rackhd-configuration) and http://rackhd.readthedocs.io/en/latest/packaging_and_deployment.html#setting-up-http-https-endpoint

I want to invite @WangWinson @cgx027 to confirm how to handle such configuration entries.